File: java/gpt/P16.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class P16 {
    public static int getByIndex(int idx) {
        int[] arr = {0,1,2,3,4,5,6,7,8,9};
        if (idx == -1) return -1;
        if (idx < 0 || idx >= arr.length) return -1;
        return arr[idx];
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Index: ");
        String s = br.readLine();
        int idx;
        try { idx = Integer.parseInt(s.trim()); } catch (Exception e) { System.out.println(-1); return; }
        System.out.println(getByIndex(idx));
    }
}

File: c/gpt/p16.c
// p16: takes integer input, returns value at that index; -1 for -1 or out-of-bounds
#include <stdio.h>

int getByIndex(int idx)
{
    int arr[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
    int n = sizeof(arr) / sizeof(arr[0]);
    if (idx == -1)
        return -1;
    if (idx < 0 || idx >= n)
        return -1;
    return arr[idx];
}

int main(void)
{
    int idx;
    printf("Index: ");
    if (scanf("%d", &idx) != 1)
    {
        printf("-1\n");
        return 0;
    }
    printf("%d\n", getByIndex(idx));
    return 0;
}
